File: ReferenceWRLImplementation/CSharpUnitTestProject/ParameterizedActivatableStaticClassUnitTests.cs
using Microsoft.VisualStudio.TestPlatform.UnitTestFramework;
using ReferenceWRLImplementation;

namespace CSharpUnitTestProject
{
    [TestClass]
    public class ParameterizedActivatableStaticClassUnitTests
    {
        [TestMethod]
        public void DefaultConstructorTest()
        {
            var parameterizedActivatableStaticClass = new ParameterizedActivatableStaticClass();
        }

        [TestMethod]
        public void ParamConstructorTest()
        {
            uint expectedVal = 45;
            var parameterizedActivatableStaticClass = new ParameterizedActivatableStaticClass(expectedVal);
            Assert.AreEqual(expectedVal, parameterizedActivatableStaticClass.NonStaticMethod());
        }

        [TestMethod]
        public void StaticMethodTest()
        {
            uint expectedVal = 40;
            var actualVal = ParameterizedActivatableStaticClass.StaticMethod();
            Assert.AreEqual(expectedVal, actualVal);
        }

        [TestMethod]
        public void NonStaticMethodTest()
        {
            uint expectedVal = 50;
            var parameterizedActivatableStaticClass = new ParameterizedActivatableStaticClass(expectedVal);
            var actualVal = parameterizedActivatableStaticClass.NonStaticMethod();
            Assert.AreEqual(expectedVal, actualVal);
        }
    }
}

File: ReferenceWRLImplementation/CXUnitTestApp/ParameterizedActivatableStaticClassUnitTests.cpp
#include "pch.h"
#include "CppUnitTest.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace ReferenceWRLImplementation;

namespace CXUnitTestApp
{
    TEST_CLASS(ParameterizedActivatableStaticClassCPPUnitTests)
    {
    public:
        TEST_METHOD(DefaultConstructorCPPTest)
        {
            auto parameterizedActivatableStaticClass = ref new ParameterizedActivatableStaticClass();
        }

        TEST_METHOD(ParamConstructorCPPTest)
        {
            UINT32 expectedVal = 45;
            auto parameterizedActivatableStaticClass = ref new ParameterizedActivatableStaticClass(expectedVal);
            Assert::AreEqual(expectedVal, parameterizedActivatableStaticClass->NonStaticMethod());
        }

        TEST_METHOD(StaticMethodCPPTest)
        {
            UINT32 expectedVal = 40;
            auto actualVal = ParameterizedActivatableStaticClass::StaticMethod();
            Assert::AreEqual(expectedVal, actualVal);
        }

        TEST_METHOD(NonStaticMethodCPPTest)
        {
            UINT32 expectedVal = 50;
            auto parameterizedActivatableStaticClass = ref new ParameterizedActivatableStaticClass(expectedVal);
            auto actualVal = parameterizedActivatableStaticClass->NonStaticMethod();
            Assert::AreEqual(expectedVal, actualVal);
        }
    };
}